.github/workflows/pr.yml

```diff
@@ -98,3 +98,93 @@ jobs:
         uses: actions/upload-artifact@v4
         with:
           path: ./**/*.apk
+
+  instrumentation_test:
+    name: "Instrumentation Tests"
+    runs-on: ubuntu-latest
+    strategy:
+      # We want the result of each device to be reported, so we can't fail-fast
+      fail-fast: false
+      matrix:
+        api-level: [21, 35]
```

Comment:
I think we should at least test the two extreme versions at the moment. In the future if we see that we need more we can add a new one.